data/src/isupport.rs

```diff
@@ -0,0 +1,614 @@
+            return Err("empty ISUPPORT parameter not allowed");
+        }
+
+        match isupport.chars().next() {
+            Some('-') => Ok(Parameter::Negation(isupport[1..].to_string())),
+            _ => {
+                if let Some((parameter, value)) = isupport.split_once('=') {
+                    match parameter {
+                        "ACCEPT" => Ok(Parameter::ACCEPT(parse_required_positive_integer(value)?)),
+                        "ACCOUNTEXTBAN" => {
+                            let account_based_extended_ban_masks =
+                                value.split(',').map(String::from).collect::<Vec<_>>();
+
+                            if !account_based_extended_ban_masks.is_empty() {
+                                Ok(Parameter::ACCOUNTEXTBAN(account_based_extended_ban_masks))
+                            } else {
+                                Err("no valid account-based extended ban masks")
+                            }
+                        }
+                        "AWAYLEN" => {
+                            Ok(Parameter::AWAYLEN(parse_optional_positive_integer(value)?))
+                        }
+                        "BOT" => Ok(Parameter::BOT(parse_required_letter(value, None)?)),
+                        "CALLERID" => Ok(Parameter::CALLERID(parse_required_letter(
+                            value,
+                            Some(default_caller_id_letter()),
+                        )?)),
+                        "CASEMAPPING" => match value {
+                            "ascii" => Ok(Parameter::CASEMAPPING(CaseMap::ASCII)),
+                            "rfc1459" => Ok(Parameter::CASEMAPPING(CaseMap::RFC1459)),
+                            "rfc1459-strict" => Ok(Parameter::CASEMAPPING(CaseMap::RFC1459_STRICT)),
+                            "rfc7613" => Ok(Parameter::CASEMAPPING(CaseMap::RFC7613)),
+                            _ => Err("unknown casemapping"),
+                        },
+                        "CHANLIMIT" => {
+                            let mut channel_limits = vec![];
+
+                            value.split(',').for_each(|channel_limit| {
+                                if let Some((prefix, limit)) = channel_limit.split_once(':') {
+                                    if limit.is_empty() {
+                                        channel_limits.push(ChannelLimit {
+                                            prefix: prefix.to_string(),
+                                            limit: None,
+                                        });
+                                    } else if let Ok(limit) = limit.parse::<u16>() {
+                                        channel_limits.push(ChannelLimit {
+                                            prefix: prefix.to_string(),
+                                            limit: Some(limit),
+                                        });
+                                    }
+                                }
```

Comment:
`prefix` should be `chars()` and each distinct channel prefix character should be stored. Each character being a separate `CHANTYPE` 

https://modern.ircdocs.horse/#chanlimit-parameter